rust/.cargo/config.toml

```diff
@@ -15,5 +15,8 @@ rustflags = [
     "-Wclippy::use_self",
 ]
 
+[target.x86_64-unknown-linux-gnu]
+rustflags = ["-C", "target-cpu=haswell", "-C", "target-feature=+avx2"]
+
 [target.aarch64-apple-darwin]
-# rustflags = ["-C", "target-cpu=native", "-C", "target-feature=+neon"]
+rustflags = ["-C", "target-cpu=apple-m1", "-C", "target-feature=+neon"]
```

Comment:
What about m2?

Reply:
This flag is for supporting 'any cpu after the target'. Similar to the x86_64 target setting to `haswell` which was a 2013 CPU, whichs that we dont support any CPUs older than that.